apps/arena/native/physics/src/collision_detection/ear_clipping.rs

```diff
@@ -0,0 +1,197 @@
+use crate::map::{Entity, Position};
+
+/*
+Ear clipping triangulation algorithm
+
+The logic of the algorithm goes like this:
+1. Traverse the polygon looking for ears
+2. Once we find an ear do the following:
+   - Save the triangle formed by the current previous and next vertex
+   - Delete the current vertex from the available vertex list
+   - Start again with the cleaned up vertex list until the list length is 3
+3. Make a final triangle with the 3 remaining vertices
+
+Even tho the resolution is different the logic is the same as the following article:
+- [Geometric tools article](https://www.geometrictools.com/Documentation/TriangulationByEarClipping.pdf)
+
+*/
+
+pub(crate) fn maybe_triangulate_polygon(mut polygon: Entity) -> Vec<Entity> {
+    clean_extra_vertices(&mut polygon);
+    if is_polygon_concave(&polygon) {
+        return vec![polygon.clone()];
+    }
+    triangulate_polygon(&polygon)
+}
+
+// Here we'll remove any vertices are shouldn't be part of the polygon or are useless
+// - If three vertices are in a line that's a useless connections that we should remove
+fn clean_extra_vertices(polygon: &mut Entity) {
+    let mut cleaned_vertices = vec![];
+    for current_vertex_index in 0..polygon.vertices.len() {
+        let previous_vertex_index =
+            get_previous_cyclic_index(current_vertex_index, &polygon.vertices);
+        let next_vertex_index = get_next_cyclic_index(current_vertex_index, &polygon.vertices);
+
+        let previous_vertex = polygon.vertices[previous_vertex_index];
+        let current_vertex = polygon.vertices[current_vertex_index];
+        let next_vertex = polygon.vertices[next_vertex_index];
+
+        if !is_vertex_consecutive(&previous_vertex, &current_vertex, &next_vertex) {
+            cleaned_vertices.push(current_vertex);
+        }
+    }
+
+    polygon.vertices = cleaned_vertices;
+}
+
+fn is_vertex_consecutive(
+    previous_vertex: &Position,
+    current_vertex: &Position,
+    next_vertex: &Position,
+) -> bool {
+    let first_vector = Position::sub(previous_vertex, current_vertex);
+    let second_vector = Position::sub(next_vertex, current_vertex);
```

Comment:
I think we should avoid using "first", "second". Maybe "previous_distance_between_vertices" and "current_distance_between_vertices" is a more descriptive option.
Also, that option seems long. Maybe we should give a name to the straight line linking two vertices and write it down somewhere. You're using "vector" for that but it's too generic.

Please give it a minute or two to come up with better variable naming, taking into account the things I mentioned.

Reply:
i'll use the same naming convention as `triangle_has_point_inside` to name this variable where we describe the positions that form the vector and the direction "current_to_previous_vector" for example